capa/ida/plugin/form.py

```diff
@@ -41,6 +41,13 @@
 CAPA_SETTINGS_RULEGEN_AUTHOR = "rulegen_author"
 CAPA_SETTINGS_RULEGEN_SCOPE = "rulegen_scope"
 
+from enum import Enum
+
+
+class Options(Enum):
```

Comment:
im just learning about this myself, but i think the appropriate class here is `Flag`:
https://docs.python.org/3/library/enum.html#flag

Reply:
Actually `(option & Options.ANALYZE) == Options.ANALYZE` (the current code) doesn't work:
```
TypeError: unsupported operand type(s) for &: 'int' and 'Options'
```

I think we should use `IntFlag`. With `class Options(Flag)`, we need to cast `options` as `&` is not defined for `int` and `Flag` and we receive and `int`: `(Options(option) & Options.ANALYZE) == Options.ANALYZE`. With `class Options(IntFlag)` the casting is not needed.

We could also use `IntEnum`. I think the difference between `IntFlag` and `IntEnum` is that it allow us to define options from other options `(VERBOSE_ANALYZE = VERBOSE | ANALYZE)`. I guess it doesn't hurt to use `IntFlag` in case we want to use it as some point.

Reply:
ok, thats fine - whatever works for our situation. i was just googling around for supported solutions to bitflags in python